### DISCUSSION TITLE
Start AR when scanning QR?

### DISCUSSION THREAD

**Discussion by danilokrulj** [google/model-viewer#4199] [Q&A]
Is it possible to enter AR when scanning a QR code without additional interaction from the user?

I made a configurator that the user can use on desktop and when he is finished configurating he gets a QR code to scan which opens the same page on mobile and applies the changes made on desktop. Can I make it so that when the QR is scanned it starts the AR experience without the user clicking a button to enter AR?

I remember that someone already said that there has to be a user interaction to start AR but I want to check if anyone had luck with something similar.

**Comment by elalish** [comment]
Indeed, it's explicitly not allowed for WebXR, and even if you can get it to work for another AR mode, I bet Chrome will consider that a bug. Honestly, hitting the user with a camera permission prompt before you even show them the object is not a great idea anyway. We have some UXR showing that loses a lot of users.

**Reply by danilokrulj** [reply]
I would like to use it in that specific case when the user configures the model on desktop and wants to switch to mobile to view it in AR.

This is an example of how I would like the AR to get triggered:
https://repfitness.com/products/pr-5000-rack-builder

**Reply by elalish** [reply]
They are directly launching a scene-viewer intent, and specifying to go into AR instead of 3D first, which means the first thing the user sees is a camera permission prompt. You're welcome to try this, but we found most users simply exit the experience rather than accepting a prompt without context.

**Comment by danilokrulj** [comment]
If anyone has a solution to this or some type of workaround I would love to hear from them how they did it. :) Or if someone has a clue how they did it in the example I shared above.

**Comment by danilokrulj** [comment]
Any breakthroughs on this topic?

**Reply by danilokrulj** [reply]
Update: 
modelViewer.activateAR() works for scene-viewer and quicklook but not for WebXR, but the problem I am having is that scene-viewer cant show changes made on the model, it can just show the original model in AR. I would like to show a configurated model in WebXR after a user configures it on desktop and scans a QR code to open and view it in AR on mobile. Any help is greatly appreciated

**Reply by elalish** [reply]
Sorry, user activation is a security requirement for WebXR.
